src/github.com/getlantern/flashlight/client/reverseproxy.go

```diff
@@ -18,7 +18,9 @@ import (
 func (client *Client) newReverseProxy(bal *balancer.Balancer) *httputil.ReverseProxy {
 	transport := &http.Transport{
 		TLSHandshakeTimeout: 40 * time.Second,
+		MaxIdleTime:         30 * time.Second,
```

Comment:
This requires our custom fork of go.  It prevents us from trying to reuse connections that timed out due to idleness.